src/clinicaltrialsgov/index.js

```diff
@@ -366,9 +329,23 @@ const upload = async ({ conn }) => {
         error: 0, success: 0,
     };
 
+    let processCount = 1,
+        total;
+
+    if (maxRecords) {
+        total = maxRecords;
+    } else {
+        total = trials.totalCount;
+    }
+
     for (const trial of trials.studies) {
```

Comment:
Why a loop here and then the same loop again at line 380?

Reply:
Put them in one loop now.